### PR TITLE
Updating docs for qiskit-transpiler-service

### PR DESCRIPTION
**Pull request by cbjuan**
Updating the documentation to reflect some of the changes we've implemented for the beta version of the service and to include citation reference

### CODE REVIEW

File: docs/transpile/qiskit-transpiler-service.mdx
Comment (Eric-Arellano) [inline docs/transpile/qiskit-transpiler-service.mdx]:
```suggestion
To get more info about the `qiskit-transpiler-service` available methods, check out the [qiskit-transpiler-service API reference](/api/qiskit-transpiler-service). To know more about the service API, check out the [Qiskit Transpiler Service REST API documentation](/api/qiskit-transpiler-service-rest)
```

**Review by Eric-Arellano — COMMENTED** [review]
Thanks!

File: docs/transpile/ai-transpiler-passes.mdx
Comment (cbjuan) [reply docs/transpile/ai-transpiler-passes.mdx:106]: Thanks. Done in https://github.com/Qiskit/documentation/pull/1435/commits/0283aff35b6a75ec118f072c52c17e2895245fc4

File: docs/transpile/ai-transpiler-passes.mdx
Comment (beckykd) [inline docs/transpile/ai-transpiler-passes.mdx:108]:
```suggestion
If you use any AI feature from the Qiskit transpiler service in your research, use [the recommended citation](./qiskit-transpiler-service#citation).
```

Comment (Eric-Arellano) [reply docs/transpile/ai-transpiler-passes.mdx:108]: > I'm not sure what the tone is supposed to be.

We should be using a consistent tone for all docs

Comment (cbjuan) [reply docs/transpile/ai-transpiler-passes.mdx]: Updated to be more friendly

File: docs/transpile/ai-transpiler-passes.mdx
Comment (Eric-Arellano) [inline docs/transpile/ai-transpiler-passes.mdx:108]: This was a bad suggestion - we want to use a consistent professional tone in our docs

```suggestion
If you use any AI feature from the Qiskit transpiler service in your research, use [the recommended citation](./qiskit-transpiler-service#citation).
```

File: docs/transpile/qiskit-transpiler-service.mdx
Comment (cbjuan) [reply docs/transpile/qiskit-transpiler-service.mdx:85]: I do agree we could do it with any `QuantumCircuit`, yet we want to showcase from the beginning this works for big circuits, that's the reason for that particular one

**Review by Eric-Arellano — APPROVED** [review]

**Comment by Eric-Arellano** [comment]
@cbjuan I've invited you with write access. You can merge this when you're ready, and then I'll deploy the docs.

**Comment by abbycross** [comment]
Sorry - I'm just back from being out sick and want to give this a pass. Will merge asap I promise! I wonder if we should be saying "AI-powered synthesis passes" rather than "AI synthesis passes", per @JulesMurphy 's caution not to leave AI unmodified -- thoughts???

File: docs/transpile/qiskit-transpiler-service.mdx
Comment (abbycross) [inline docs/transpile/qiskit-transpiler-service.mdx:29]:
```suggestion
*Note*: This package requires Qiskit SDK v1.X.
```

File: docs/transpile/qiskit-transpiler-service.mdx
Comment (abbycross) [inline docs/transpile/qiskit-transpiler-service.mdx:36]:
```suggestion
- `ai` ("true", "false", "auto") - Whether to use AI-powered capabilities during transpilation. The AI-powered capabilities available can be for `AIRouting` transpiling passes or other AI synthesis methods. If this value is `"true"`, the service applies different AI-powered transpiling passes depending on the `optimization_level` requested. If  `"false"`, it uses the latest Qiskit transpiling features without AI. Finally, if `"auto"`, the service decides whether to apply the standard Qiskit heuristic passes or the AI-powered passes based on your circuit.
```

File: docs/transpile/ai-transpiler-passes.mdx
Comment (abbycross) [inline docs/transpile/ai-transpiler-passes.mdx:108]:
```suggestion
If you use any AI-powered feature from the Qiskit transpiler service in your research, use [the recommended citation](./qiskit-transpiler-service#citation).
```

File: docs/transpile/qiskit-transpiler-service.mdx
Comment (abbycross) [inline docs/transpile/qiskit-transpiler-service.mdx:97]:
```suggestion
If you use any AI-powered feature from the Qiskit transpiler service in your research, use the following recommended citation:
```

File: docs/transpile/ai-transpiler-passes.mdx
Comment (abbycross) [inline docs/transpile/ai-transpiler-passes.mdx:103]:
```suggestion
These custom collection passes limit the sizes of the collected sub-circuits so they are supported by the AI-powered synthesis passes. Therefore, it is recommended to use them after the routing passes and before the synthesis passes for a better overall optimization.
```

**Comment by cbjuan** [comment]
Yes, that's right @abbycross. In general we should say always AI-powered. Thanks for the suggestion and for jumping in right on time :)

File: docs/transpile/qiskit-transpiler-service.mdx
Comment (abbycross) [inline docs/transpile/qiskit-transpiler-service.mdx:36]:
```suggestion
- `ai` ("true", "false", "auto") - Whether to use AI-powered capabilities during transpilation. The AI-powered capabilities available can be for `AIRouting` transpiling passes or other AI-powered synthesis methods. If this value is `"true"`, the service applies different AI-powered transpiling passes depending on the `optimization_level` requested. If  `"false"`, it uses the latest Qiskit transpiling features without AI. Finally, if `"auto"`, the service decides whether to apply the standard Qiskit heuristic passes or the AI-powered passes based on your circuit.
```

**Review by abbycross — APPROVED** [review]

**Pull request merged**
